metomi/rosie/suite_id.py

```diff
@@ -435,46 +436,22 @@ def _find_vc_file_from_location(location):
         return None
 
     @staticmethod
-    def _parse_cylc_vc_file(fpath):
+    def _parse_cylc_vc_file(fpath: str) -> Optional[str]:
         """Take a path to a Cylc VC file and returns an svn URL.
 
         Args:
             fpath: Location of Cylc Version Control log file.
 
-        Returns: SVN location, or None
-
-        Examples:
-            >>> class MockFpath():
-            ...     def read_text(self):
-            ...         return (
-            ...             'version control system="svn"\\n'
-            ...             'url="/a/b/c"\\n'
-            ...             'revision="4242"'
-            ...         )
-            >>> mypath = MockFpath()
-            >>> SuiteId.parse_cylc_vc_file(mypath)
-            '/a/b/c@4242'
```

Comment:
I couldn't come up with a concise way to mock `open()`, so have just ditched this test :/

Turns out the doctest won't get run normally anyway: https://github.com/metomi/rose/blob/19e6c87bf6045950d42e684219ec22ee6656e880/pytest.ini#L1-L5

Reply:
Hmm, not sure why, this diff fixes it:

```diff
diff --git a/metomi/rosie/suite_id.py b/metomi/rosie/suite_id.py
index f9f75e43..dc086e20 100644
--- a/metomi/rosie/suite_id.py
+++ b/metomi/rosie/suite_id.py
@@ -452,7 +452,7 @@ class SuiteId:
             ...             'revision="4242"'
             ...         )
             >>> mypath = MockFpath()
-            >>> SuiteId.parse_cylc_vc_file(mypath)
+            >>> SuiteId._parse_cylc_vc_file(mypath)
             '/a/b/c@4242'
         """
         location = None
diff --git a/pytest.ini b/pytest.ini
index 94c8001f..adbd85d0 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -2,9 +2,7 @@
 addopts =
     --verbose
     --doctest-modules
-    --ignore=metomi/rosie
     --ignore=metomi/rose/ws.py
-    --ignore=metomi/rose/metadata_graph.py
     # these cause import issues
     --ignore=metomi/rose/etc/
     # these tests do IO, don't run them under sphinx-build rather than pytest:
```

Reply:
There were a couple of import errors so I've added `--ignore`s for those files. I've replaced the doctest with a unit test. https://github.com/metomi/rose/pull/2520/commits/55646554de85607b53e16a0700e6de0d2680731c

Reply:
Hmm, I didn't get any import issues?